src/NServiceBus.Persistence.AzureStorage/Timeout/TimeoutLogic/TimeoutPersister.cs

```diff
@@ -158,25 +158,23 @@ public TimeoutPersister(string timeoutConnectionString, string timeoutDataTableN
             var lastSuccessfulReadEntity = await GetLastSuccessfulRead(timeoutManagerDataTable).ConfigureAwait(false);
             var lastSuccessfulRead = lastSuccessfulReadEntity?.LastSuccessfullRead;
 
-            TableQuery<TimeoutDataEntity> query;
+            var floor = lastSuccessfulRead ?? DateTime.UtcNow.AddYears(-1);
 
-            if (lastSuccessfulRead.HasValue)
-            {
-                query = new TableQuery<TimeoutDataEntity>()
-                    .Where(
+            var query = new TableQuery<TimeoutDataEntity>()
+                .Where(
+                    TableQuery.CombineFilters(
                         TableQuery.CombineFilters(
-                            TableQuery.CombineFilters(TableQuery.GenerateFilterCondition("PartitionKey", QueryComparisons.GreaterThanOrEqual, lastSuccessfulRead.Value.ToString(partitionKeyScope)),
-                                TableOperators.And,
-                                TableQuery.GenerateFilterCondition("PartitionKey", QueryComparisons.LessThanOrEqual, now.ToString(partitionKeyScope))),
+                            TableQuery.GenerateFilterCondition("PartitionKey", QueryComparisons.GreaterThanOrEqual, floor.ToString(partitionKeyScope)),
```

Comment:
With a single endpoint using a storage account this isn't much help, since when `lastSucessfullRead` is null there really shouldn't be any timeouts for that endpoint. 

But when multiple endpoints are sharing the storage account, and a new endpoint starts using it, the primary key culling will speed up the query so that all the partitions need not be scanned for the `OwningTimeoutManager` condition below.